src/uct/tcp/tcp_iface.c

```diff
@@ -772,10 +768,14 @@ ucs_status_t uct_tcp_query_devices(uct_md_h md,
             continue;
         }
 
-        uct_md_iface_config_read(md, "tcp", NULL, NULL, &config);
-        iface_config = ucs_derived_of(config, uct_tcp_iface_config_t);
+        uct_md_config_read(md->component, "tcp", NULL, &md_config);
```

Comment:
should take it from tcp_md object (fill it when initializing tcp md), instead of read every time during query_resources